custom_components/healthchecksio/__init__.py

```
@@ -114,28 +137,47 @@ def __init__(self, hass, api_key, check, self_hosted, site_root, ping_endpoint):
+        timeout10 = aiohttp.ClientTimeout(total=10)
+        headers = {"X-Api-Key": self.api_key}
+        if self.check is not None:
+            if self.self_hosted:
+                check_url = f"{self.site_root}/{self.ping_endpoint}/{self.check}"
+            else:
+                check_url = f"https://hc-ping.com/{self.check}"
+            await asyncio.sleep(1)  # needed for self-hosted instances
+            try:
+                check_response = await session.get(check_url, timeout=timeout10)
+            except (aiohttp.ClientError, asyncio.TimeoutError) as error:
+                _LOGGER.error(f"Could Not Send Check: {error}")
```

Ruff (0.11.9)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)